README.md

```diff
@@ -1,67 +1,90 @@
 
+### Basic
 
-- Windows
-    - Install [VSCode](https://code.visualstudio.com/) (optionally)
-    - Install [Rust](https://www.rust-lang.org/learn/get-started)
-    - Install [MSYS2](https://www.msys2.org/)
-    - Cortex-M7F with hardware floating point (ARMv7E-M architecture):
-        ```bash
-        rustup target add thumbv7em-none-eabihf
-        ```
-    - cargo-binutils
-        ```bash
-        cargo install cargo-binutils
-        rustup component add llvm-tools-preview
-        ```
-    - cargo-generate
-        ```bash
-        cargo install cargo-generate
-        ```
-    - arm-none-eabi-gdb
-        ```bash
-        pacman -S gcc
-        pacman -S mingw-w64-x86_64-gdb
-        pacman -S mingw-w64-x86_64-arm-none-eabi-gdb
-        ```
-        or download [here](https://developer.arm.com/downloads/-/gnu-rm) and install it
-    - OpenOCD
-        ```bash
-        pacman -S mingw-w64-x86_64-openocd
-        ```
-- Linux (Ubuntu 22.04)
+#### Prerequisites
+none
+
+#### Scope
+Build and Flash the Firmware
 
-    - Install Rust using command below and follow the on-screen instructions
+#### Steps
+- Install Rust
+    - Linux (Ubuntu 22.04)
+
+        *Run the command below and follow the on-screen instructions*
         ```bash
         curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
         ```
-    - Cortex-M7F with hardware floating point (ARMv7E-M architecture):
-        ```bash
-        rustup target add thumbv7em-none-eabihf
-        ```
-    - cargo-binutils
         ```bash
-        cargo install cargo-binutils
-        rustup component add llvm-tools-preview
+        sudo apt install libudev-dev
         ```
-    - cargo-generate
+    - Windows
+
+        Download and Install Rust from [here](https://www.rust-lang.org/tools/install)
+    - macOS
+
+        [Update Homebrew](https://docs.brew.sh/FAQ)
         ```bash
-        cargo install cargo-generate
+        brew update
+        brew upgrade
         ```
-    - cargo-flash
         ```bash
-        sudo apt install libudev-dev
-        cargo install cargo-flash
+        brew install rust
         ```
-    - gcc-arm-embedded
+- Add Cortex-M7 with hardware floating point (ARMv7E-M architecture):
+    ```bash
+    rustup target add thumbv7em-none-eabihf
+    ```
+- cargo-binutils
+    ```bash
+    cargo install cargo-binutils
+    rustup component add llvm-tools-preview
+    ```
+- cargo-generate
+    ```bash
+    cargo install cargo-generate
+    ```
+- cargo-flash
+    ```bash
+    cargo install cargo-flash
+    ```
+- ST-LINK USB driver
+    - Windows
+
+        Download and install it from [here](https://www.st.com/en/development-tools/stsw-link009.html)
+
+### Complete
+
+#### Prerequisites
+Finish [Basic](#basic) setup
+
+#### Scope
+Design, Development, Testing, Debugging, etc.
+
+#### Steps
+- [Git](https://git-scm.com/downloads)
+- [VSCode](https://code.visualstudio.com/)
```

Comment:
The documentation looks good now, but at least for this VSCode part you might want to highlight it is for Windows only.
Generally speaking, I think we need complete but separate guides (from beginning to end) for Windows and Linux environments, but this is a task by itself.

Reply:
VsCode is available for Linux environments, too. But, I also support the fact that we should not be bound by a specific IDE. Maybe we can put in place a package a dependency and environment management (package management) like Conda or Conan and create multiple environments customized to smaller groups of people.

Reply:
As @DanDragan said, VSCode indeed is cross-OS compatible, but of course it is not mandatory but rather recommended (I will add a mention about it).
Regarding separate guidelines for each OS, we had it like this previously, but I decided to go into a more hybrid approach.
First I separated Setup into `Basic` and `Complete` chapters and then listed all necessary tools that go in each chapter. For some tools, there might be OS specific setup.